tasks/pdp/task_terminate_data_set.go

```diff
@@ -0,0 +1,205 @@
+	}
+
+	live, err := verifier.DataSetLive(&bind.CallOpts{Context: ctx}, big.NewInt(dataSetId))
+	if err != nil {
+		return false, xerrors.Errorf("failed to check if data set is live: %w", err)
+	}
+
+	if !live {
+		n, err := t.db.Exec(ctx, `UPDATE pdp_delete_data_set SET 
+                               after_delete_data_set = TRUE,
+                               delete_data_set_task_id = NULL,
+                               terminated  = TRUE
+                           WHERE terminate_service_task_id = $1`, taskID)
+		if err != nil {
+			return false, xerrors.Errorf("failed to update pdp_delete_data_set: %w", err)
+		}
+
+		if n != 1 {
+			return false, xerrors.Errorf("expected to update 1 row but got %d", n)
+		}
+
+		return true, nil
+	}
+
+	pdpABi, err := contract.PDPVerifierMetaData.GetAbi()
+	if err != nil {
+		return false, xerrors.Errorf("failed to get PDPVerifier ABI: %w", err)
+	}
+
+	data, err := pdpABi.Pack("deleteDataSet", big.NewInt(dataSetId), []byte{})
+	if err != nil {
+		return false, xerrors.Errorf("failed to pack data: %w", err)
+	}
+
+	txEth := types.NewTransaction(
+		0,
+		pdpAddress,
+		big.NewInt(0),
+		0,
+		nil,
+		data,
+	)
+
+	txHash, err := t.sender.Send(ctx, sender, txEth, "pdp-terminate-data-set")
+	if err != nil {
+		return false, xerrors.Errorf("failed to send transaction: %w", err)
+	}
+
+	n, err := t.db.Exec(ctx, `UPDATE pdp_delete_data_set SET 
+                               delete_tx_hash = $2, 
+                               after_delete_data_set = TRUE,
+                               delete_data_set_task_id = NULL
+                           WHERE terminate_service_task_id = $1`, taskID, txHash.Hex())
+	if err != nil {
+		return false, xerrors.Errorf("failed to update pdp_delete_data_set: %w", err)
+	}
+
+	if n != 1 {
+		return false, xerrors.Errorf("expected to update 1 row but got %d", n)
+	}
+
+	return true, nil
+}
+
+func (t *DeleteDataSetTask) CanAccept(ids []harmonytask.TaskID, engine *harmonytask.TaskEngine) (*harmonytask.TaskID, error) {
+	return &ids[0], nil
+}
+
+func (t *DeleteDataSetTask) TypeDetails() harmonytask.TaskTypeDetails {
+	return harmonytask.TaskTypeDetails{
+		Name: "DeleteDataSet",
+		Cost: resources.Resources{
+			Cpu: 1,
+			Gpu: 0,
+			Ram: 64 << 20,
+		},
+		MaxFailures: 3,
+		IAmBored: passcall.Every(time.Hour, func(taskFunc harmonytask.AddTaskFunc) error {
+			return t.schedule(context.Background(), taskFunc)
+		}),
+	}
+}
+
+func (t *DeleteDataSetTask) schedule(ctx context.Context, addTaskFunc harmonytask.AddTaskFunc) error {
+	var stop bool
+
+	for !stop {
+		addTaskFunc(func(taskID harmonytask.TaskID, tx *harmonydb.Tx) (shouldCommit bool, seriousError error) {
+			stop = true
+
+			current, err := t.ethClient.BlockNumber(ctx)
+			if err != nil {
+				return false, xerrors.Errorf("failed to get current block number: %w", err)
+			}
+
+			var pendings []struct {
+				ID               int64 `db:"id"`
+				TerminationEpoch int64 `db:"termination_epoch"`
```

Comment:
nit: repeating myself -- `service_termination_epoch` would help make this a little more clear